community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -226,6 +227,41 @@ else if ( seconds > 0 && nanos < 0 )
         this.nanos = (int) nanos;
     }
 
+    public int compareTo( DurationValue other )
+    {
+        return Comparator.comparingLong(DurationValue::averageLengthInNanos)
+                .thenComparingLong( DurationValue::getMonths )
+                .thenComparingLong( DurationValue::getDays )
+                .thenComparingLong( DurationValue::getSeconds )
+                .thenComparingInt( DurationValue::getNanos ).compare( this, other );
```

Comment:
Tiny implementation note: the Comparator should be statically defined, it is unnecessary to recreate a new Comparator every time we compare two DurationValue objects.